src/main/java/ch/njol/skript/classes/data/DefaultComparators.java

```diff
@@ -630,6 +631,19 @@ public boolean supportsOrdering() {
 				return false;
 			}
 		});
+
+		// Color - Color
+		Comparators.registerComparator(Color.class, Color.class, new Comparator<Color, Color>() {
+			@Override
+			public Relation compare(Color color1, Color color2) {
+				return Relation.get(color1.getName().equals(color2.getName()));
+			}
+
+			@Override
+			public boolean supportsOrdering() {
+				return false;
+			}
+		});
```

Comment:
This should not be needed, Skript should be sending through https://github.com/SkriptLang/Skript/blob/0c4a190b0d1d54a0a5ce4a81127b744cefd38cc9/src/main/java/ch/njol/skript/classes/data/DefaultComparators.java#L484

Reply:
I will revisit this to double check but as I remember, there was an issue where it says `Can't compare color with color` which is why I used this to fix it.

Reply:
This is probably the same issue discussed by Fuzesion in the discord before, the comparator right now needs to be registered due to a bug in the if statement, you may register it and plan for removal upon that bug is fixed if anything.

Reply:
To provide feedback on this the issue is `if colored of %object% is %color%` is what does this, it's an issue I've mentioned in skriptlang-development a week or so back now it's related to expression comparing against literals, this should be retained and a proper fix should be done to CondCompare if we could finally find why it is failing.

Reply:
I removed the quick fix in favor to a proper fix in CondCompare